### ISSUE TITLE
Psalm silently ignores duplicate `@return` comments

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#2245]
Observed: It uses the first comment and ignores the second
Expected: Combine the union types of all `@return` statements (continue to have psalm-return take precedence), or warn about multiple return comments being unsupported

```php
<?php

class X {
    /**
     * @return string in some conditions
     * @return array in some other conditions
     */
    function test() {
        // Observed: InvalidReturnType
        // Expected: No warning
        return [];
    }
}
```

**Comment by SignpostMarv** [comment]
related to #1400 ?
